records/labels/tables_test.go

```diff
@@ -0,0 +1,103 @@
+package labels
+
+import (
+	"testing"
+)
+
+func BenchmarkAsDNS952(b *testing.B) {
+	const (
+		original = "89f.gsf---g_7-fgs--d7fddg-"
+		expected = "f-gsf---g-7-fgs--d7fddg"
+	)
+
+	asDNS952 := HostNameSpec952.Mangler().Mangle
```

Comment:
I think it'd be really nice if HostNameSpec would implement the Mangler interface directly.